compact.go

```diff
@@ -520,20 +536,33 @@ func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockRe
 	if err != nil {
 		return errors.Wrap(err, "open index writer")
 	}
-	defer indexw.Close()
+	writers = append(writers, indexw)
 
 	if err := c.populateBlock(blocks, meta, indexw, chunkw); err != nil {
 		return errors.Wrap(err, "write compaction")
 	}
+
+	// Compaction was canceled so remove tmp folders and return early.
+	select {
+	case <-c.ctx.Done():
+		for _, w := range writers {
```

Comment:
Hm.. this is something we need to improve. There is literally no need to close those twice. It is just enough to `return ErrCompactionCanceled` here and have nice defer that will always close & remove tmp.

Reply:
updated.